src/components/Alerting.tsx

```diff
@@ -156,9 +156,20 @@ export const Alerting: FC = () => {
           </Button>
         </div>
       )}
-      {alertingRules.map((alertRule, index) => (
-        <AlertRuleForm key={`${alertRule.alert}-${index}`} rule={alertRule} onSubmit={getUpdateRules(index)} />
-      ))}
+      {alertingRules.map((rule, index) => {
+        switch (alertFamilyFromRule(rule)) {
+          case AlertFamily.ProbeDuration:
+            return <ProbeDurationAlertForm rule={rule} onSubmit={getUpdateRules(index)} />;
+
+          case AlertFamily.ProbeSuccess:
+            return <AlertRuleForm key={`${rule.alert}-${index}`} rule={rule} onSubmit={getUpdateRules(index)} />;
+
+          case AlertFamily.SSLCertExpiry:
+            return <SSLCertExpiryAlertForm rule={rule} onSubmit={getUpdateRules(index)} />;
+        }
+
+        return undefined;
+      })}
```

Comment:
I'm sure there's a better way...

Reply:
Not crazy about the need for a `switch` here but it seems like the logic is pretty intense

Reply:
the switch is gone now...